webapp/packages/plugin-connections-administration/src/Administration/Connections/ConnectionsTable/ConnectionsTable.tsx

```diff
@@ -44,16 +30,17 @@ export const ConnectionsTable = observer<Props>(function ConnectionsTable({ keys
         <TableColumnHeader min />
         <TableColumnHeader>{translate('connections_connection_name')}</TableColumnHeader>
         <TableColumnHeader>{translate('connections_connection_address')}</TableColumnHeader>
-        {displayProjects && <TableColumnHeader>{translate('connections_connection_project')}</TableColumnHeader>}
+        {state.shouldDisplayProjects && <TableColumnHeader>{translate('connections_connection_project')}</TableColumnHeader>}
```

Comment:
It is bad practice to move all logic to a single place instead of splitting it between places where you use it. `shouldDisplayProjects` is not used outside of this component, but you keep it in a `state` shared among different components. Now, look at `useConnectionsTable`. It's almost 200 rows of code.

It's always better to keep non-reused code close to its uses. This will reduce the amount of logic in single logical units and improve the readability of code.

Reply:
same situation with `getProjectName` and `getConnectionIcon`